### DISCUSSION TITLE
Target C++14 instead of C++11

### DISCUSSION THREAD

**Discussion by mxxo** [nrc-cnrc/EGSnrc#660] [Programming]
We are planning to require C++11 to compile EGSnrc. C++11 is a huge release with a ton of useful features. C++14 on the other hand was mostly intended to fix bugs and mistakes in C++11 itself. What does everyone think about moving directly to C++14 and save ourselves the headache of strict C++11 compliance?

(Taking the argument further, why not move to C++17? Comparing feature coverage of C++14 and 17 [here](https://en.cppreference.com/w/cpp/compiler_support), we see C++17 support is a lot more fragmented than C++14 and I'm not sure what appetite there is for C++17 right away).

**Comment by ftessier** [comment]
I vote to target C++17 compliance right away. By the time we are done, ... 😉

**Reply by mxxo** [reply]
Yes, I'd also be happy to try that as well if everyone is on board 😄
